src/bin/pg_autoctl/primary_standby.c

```diff
@@ -591,8 +640,31 @@ standby_init_database(LocalPostgresServer *postgres,
 	 * Now, we know that pgdata either doesn't exists or belongs to a stopped
 	 * PostgreSQL instance. We can safely proceed with pg_basebackup.
 	 */
-	if (!pg_basebackup(pgSetup->pgdata, pgSetup->pg_ctl, replicationSource))
+	if (!pg_basebackup(pgSetup->pgdata,
+					   pgSetup->pg_ctl,
+					   replicationSource))
+	{
+		return false;
+	}
+
```

Comment:
Let's keep the `normalize_filename` call I added here, so that pgdata is fully qualified in all your calls too.

Reply:
In my copy this call is there still after the rebase, so I'm a little surprised by your comment. That said, I have moved the call to `normalize_filename` to the `local_postgres_update` function that I think it's a better place.

Reply:
What I meant was move it back to this location. I think your move of the call to `local_postgres_update` is fine, but we need to move the `join_path_components` call for the config path after it too. Otherwise it still has the relative path.

Reply:
Oh I see. Yet tests did pass, should we review the tests (in another PR) related to relative pathnames impacts?

Meanwhile I have refactored where we compute the configFilePath and done that in the `pg_setup_standby_mode` function itself: after all we have to pass PGDATA down to it, and the configFilePath is a direct compute from PGDATA.